documentation/testing/integration-tests.md

```diff
@@ -0,0 +1,201 @@
+# Integration tests
+
+The framework also provides multiple ways of integration testing.
+
+## Rib test
+
+We can create an instance of `Node` (or `Rib`) with help of `Builder` in JUnit test and verify its behaviour quite fast without using any emulators.
+To do so we need to use one of the framework features – view factory replacement.
+We can replace `View` implementation of `RIB` with stub implementation (as we did previously in `Interactor`) while keeping every other internal component intact.
+`View` implementation can be unit tested separately.
+To control the lifecycle of `Node` please use `RibNodeTestHelper`.
+
+```kotlin
+class SomeScreenNodeTest {
+
+    // Test implementations of network and analytics components, that allows to stub results and verify calls 
+    private val testTracker = TestAnalyticsTracker()
+    private val testNetwork = TestNetwork()
+
+    // We can replace view implementation with stub, so we can run this test without Robolectric
+    // RibViewStub is suitable for MVI approach, but you can easily create your own View stub just by implementing View interface
+    private val view = object : RibViewStub<SomeScreenView.ViewModel, SomeScreenView.Event>(), SomeScreenView {}
+
+    // Create the node under test through builder, provide stub dependencies
+    private val node = SomeScreenBuilder(
+        dependency = object : SomeScreen.Dependency {
+            override val network: Network = testNetwork
+            override val analyticsTracker: AnalyticsTracker = testTracker
+        }
+    ).build(
+        buildContext = BuildContext.root(
+            savedInstanceState = null,
+            // Override view implementation with stub
+            customisations = SomeScreen.Customisation { { view } }.toDirectory()
+        )
+    )
+
+    // RibNodeTestHelper will invoke all required callbacks to setup Node properly
+    private val nodeTestHelper = RibNodeTestHelper(node)
+
+    @Test
+    fun `GIVEN not loading WHEN click button THEN display loading state`() {
+        // Move Node to STARTED state
+        nodeTestHelper.moveToStateAndCheck(Lifecycle.State.STARTED) {
+
+            // Fake user actions on UI
+            view.viewEventRelay.accept(SomeScreenView.Event.ButtonPressed)
+
+            // Assert that View received correct state to render
+            assertThat(view.lastViewModel).isEqualTo(SomeScreenView.ViewModel(isLoading = true))
+        }
+    }
+
+    @Test
+    fun `GIVEN not loading WHEN input SomeInput THEN output SomeOutput`() {
+        // Observe outputs
+        val outputTest = node.output.test()
+
+        nodeTestHelper.moveToStateAndCheck(Lifecycle.State.STARTED) {
+            // Send some input
+            node.input.accept(SomeScreen.Input.SomeInput)
+
+            // Verify outputs
+            outputTest.assertValue(SomeScreen.Output.SomeOutput)
+        }
+    }
+
+}
+```
+
+In case when `Rib` has children, use the same approach as for `Interactor` and `Router` tests. 
```

Comment:
Not completely sure, but i think that "in case when" is not correct. An alternative approach could be:
"In case of Rib having children"